packages/flutter/test/widgets/implicit_animations_test.dart

```diff
@@ -171,6 +171,102 @@ void main() {
+    await tester.pumpWidget(wrap(
+      child: TestAnimatedWidget(
+        callback: mockOnEndFunction.handler,
+        switchKey: switchKey,
+        state: _TestAnimatedScaleWidgetState(),
+      ),
+    ));
+
+    final Finder widgetFinder = find.byKey(switchKey);
+
+    await tester.tap(widgetFinder);
+    await tester.pump();
+    expect(mockOnEndFunction.called, 0);
+    await tester.pump(animationDuration);
+    expect(mockOnEndFunction.called, 0);
+    await tester.pump(additionalDelay);
+    expect(mockOnEndFunction.called, 1);
+  });
+
+  testWidgets('AnimatedScale transition test', (WidgetTester tester) async {
+    await tester.pumpWidget(wrap(
+      child: TestAnimatedWidget(
+        switchKey: switchKey,
+        state: _TestAnimatedScaleWidgetState(),
+      ),
+    ));
+
+    final Finder switchFinder = find.byKey(switchKey);
+    final ScaleTransition scaleWidget = tester.widget<ScaleTransition>(
+      find.ancestor(
+        of: find.byType(Placeholder),
+        matching: find.byType(ScaleTransition),
+      ).first,
+    );
+
+    await tester.tap(switchFinder);
+    await tester.pump();
+    expect(scaleWidget.scale.value, equals(1.0));
+
+    await tester.pump(const Duration(milliseconds: 500));
+    expect(scaleWidget.scale.value, equals(1.5));
+    await tester.pump(const Duration(milliseconds: 250));
+    expect(scaleWidget.scale.value, equals(1.75));
+    await tester.pump(const Duration(milliseconds: 250));
+    expect(scaleWidget.scale.value, equals(2.0));
+  });
+
+  testWidgets('AnimatedRotation onEnd callback test', (WidgetTester tester) async {
+    await tester.pumpWidget(wrap(
+      child: TestAnimatedWidget(
+        callback: mockOnEndFunction.handler,
+        switchKey: switchKey,
+        state: _TestAnimatedRotationWidgetState(),
+      ),
+    ));
+
+    final Finder widgetFinder = find.byKey(switchKey);
+
+    await tester.tap(widgetFinder);
+    await tester.pump();
+    expect(mockOnEndFunction.called, 0);
+    await tester.pump(animationDuration);
+    expect(mockOnEndFunction.called, 0);
+    await tester.pump(additionalDelay);
+    expect(mockOnEndFunction.called, 1);
+  });
+
+  testWidgets('AnimatedRotation transition test', (WidgetTester tester) async {
+    await tester.pumpWidget(wrap(
+      child: TestAnimatedWidget(
+        switchKey: switchKey,
+        state: _TestAnimatedRotationWidgetState(),
+      ),
+    ));
+
+    final Finder switchFinder = find.byKey(switchKey);
+    final RotationTransition rotationWidget = tester.widget<RotationTransition>(
+      find.ancestor(
+        of: find.byType(Placeholder),
+        matching: find.byType(RotationTransition),
+      ).first,
+    );
+
+    await tester.tap(switchFinder);
+    await tester.pump();
+    expect(rotationWidget.turns.value, equals(0.0));
+
+    await tester.pump(const Duration(milliseconds: 500));
+    expect(rotationWidget.turns.value, equals(0.75));
```

Comment:
we should probably also verify that the widget itself _didn't_ change (i.e. that we're not rebuilding the AnimatedRotation).

also we should probably verify that the rotation is having an effect (e.g. by using the transformation matrix logic to convert a point from global coordinates to the child's coordinates).

Reply:
I wasn't sure how to reach in and verify that the AnimatedScale and AnimatedRotation widgets themselves (or their implicitly associated AnimatedState objects) were being rebuilt so I added the next best thing which is to ensure that the test harness State object that encloses them is not rebuilt. I also upgraded other tests in the same file to use the same mechanism.

With respect to verifying the actual rotation/scale of the child, since the tests verify that there is a FooTransition object underneath them and that said FooTransition object has been updated with the correct scale/rotation, wouldn't the testing that the transform is really being applied fall to the FooTransition widget tests?

Reply:
@Hixie are the new mechanisms sufficient to test your concerns?